sydtest/src/Test/Syd/Expectation.hs

```diff
@@ -83,6 +83,13 @@ shouldContain a i = shouldSatisfyNamed a ("has infix\n" <> ppShow i) (isInfixOf
 
 infix 1 `shouldContain`
 
+-- | Assert that the given list contains all elements from the other
+-- given list and only them, perhaps in a different order.
+shouldMatchList :: (HasCallStack, Show a, Eq a) => [a] -> [a] -> Expectation
```

Comment:
@sol Just to be extra sure: Are you ok with this being here under the license that it has, given that the inspiration came from hspec-expectations ?

Reply:
I have not looked at the shouldMatchList implementation, only at the API and docs. But I understand the concern

Reply:
I don't really have an opinion on this.

Reply:
I honestly didn't look at the code, but I see now how it can be legally and ethically ambiguous. I should have thought about it before